controller/controller.go

```diff
@@ -552,7 +555,18 @@ func NewProvisionController(
 	claimHandler := cache.ResourceEventHandlerFuncs{
 		AddFunc:    func(obj interface{}) { controller.enqueueWork(controller.claimQueue, obj) },
 		UpdateFunc: func(oldObj, newObj interface{}) { controller.enqueueWork(controller.claimQueue, newObj) },
-		DeleteFunc: func(obj interface{}) { controller.forgetWork(controller.claimQueue, obj) },
+		DeleteFunc: func(obj interface{}) {
+			// Do not Forget a claim if it is in claimsToRetry
+			key, err := getQueueKey(obj)
+			if err != nil {
+				utilruntime.HandleError(err)
+				return
+			}
+			if _, ok := controller.claimsToRetry.Load(key); ok {
```

Comment:
I want to say: if the claim is in the queue && deleted, keep it in the queue. But there is no easy way to tell if something is in the queue or going to be re-added to the queue. So claimsToRetry is basically serving as a copy of the queue right now...